File: ASP.NET/WebForms/VB/ConnectionHelper.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Security
Imports DevExpress.ExpressApp.Security.ClientServer
Imports DevExpress.Persistent.BaseImpl.PermissionPolicy
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Security
Imports XafSolution.Module.BusinessObjects

Namespace WebFormsApplication
    Public NotInheritable Class ConnectionHelper

        Private Sub New()
        End Sub

        Public Shared Function GetObjectSpaceProvider(ByVal security As SecurityStrategyComplex) As SecuredObjectSpaceProvider
            Dim objectSpaceProvider As New SecuredObjectSpaceProvider(security, XpoDataStoreProviderService.GetDataStoreProvider(Nothing, True), True)
            RegisterEntities(objectSpaceProvider)
            Return objectSpaceProvider
        End Function
        Public Shared Function GetSecurity(ByVal authenticationName As String, ByVal parameter As Object) As SecurityStrategyComplex
            Dim authentication As New AuthenticationMixed()
            authentication.LogonParametersType = GetType(AuthenticationStandardLogonParameters)
            authentication.AddAuthenticationStandardProvider(GetType(PermissionPolicyUser))
            authentication.AddIdentityAuthenticationProvider(GetType(PermissionPolicyUser))
            authentication.SetupAuthenticationProvider(authenticationName, parameter)
            Dim security As New SecurityStrategyComplex(GetType(PermissionPolicyUser), GetType(PermissionPolicyRole), authentication)
            security.RegisterXPOAdapterProviders()
            Return security
        End Function
        Private Shared Sub RegisterEntities(ByVal objectSpaceProvider As SecuredObjectSpaceProvider)
            objectSpaceProvider.TypesInfo.RegisterEntity(GetType(Employee))
            objectSpaceProvider.TypesInfo.RegisterEntity(GetType(PermissionPolicyUser))
            objectSpaceProvider.TypesInfo.RegisterEntity(GetType(PermissionPolicyRole))
        End Sub
    End Class
End Namespace

File: XPO/ASP.NET/WebForms/CS/ConnectionHelper.cs
using BusinessObjectsLibrary.BusinessObjects;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Security;
using DevExpress.ExpressApp.Security.ClientServer;
using DevExpress.Persistent.BaseImpl.PermissionPolicy;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Security;

namespace WebFormsApplication {
	public static class ConnectionHelper {
		public static SecuredObjectSpaceProvider GetObjectSpaceProvider(SecurityStrategyComplex security) {
			SecuredObjectSpaceProvider objectSpaceProvider = new SecuredObjectSpaceProvider(security, XpoDataStoreProviderService.GetDataStoreProvider(null, true), true);
			RegisterEntities(objectSpaceProvider);
			return objectSpaceProvider;
		}
		public static SecurityStrategyComplex GetSecurity(string authenticationName, object parameter) {
			AuthenticationMixed authentication = new AuthenticationMixed();
			authentication.LogonParametersType = typeof(AuthenticationStandardLogonParameters);
			authentication.AddAuthenticationStandardProvider(typeof(PermissionPolicyUser));
			authentication.AddIdentityAuthenticationProvider(typeof(PermissionPolicyUser));
			authentication.SetupAuthenticationProvider(authenticationName, parameter);
			SecurityStrategyComplex security = new SecurityStrategyComplex(typeof(PermissionPolicyUser), typeof(PermissionPolicyRole), authentication);
			security.RegisterXPOAdapterProviders();
			return security;
		}
		public static void RegisterEntities(IObjectSpaceProvider objectSpaceProvider) {
			objectSpaceProvider.TypesInfo.RegisterEntity(typeof(Employee));
			objectSpaceProvider.TypesInfo.RegisterEntity(typeof(PermissionPolicyUser));
			objectSpaceProvider.TypesInfo.RegisterEntity(typeof(PermissionPolicyRole));
		}
	}
}